tester.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+version="V0.0.3"
+usage="$(basename "$0") [-help] [-v] -- Simple test suite for dfshack
+
+where:
+    -h,   show this help text
+    -v,   Shows version information"
+
+###############################################################################
+#Function: auto_testing                                                       #
+#Prupose: Runs auto test case for dfshack by creating severle files changing  #
+#         their permissions and moving them around                            #
+#arguments: none                                                              #
+#return: none                                                                 #
+###############################################################################
+function auto_testing {
+  echo -e "\033[1;33mAutomated dfshack test\033[0m"
+ 
+  echo -e "\033[1;35mCreating folder outer\033[0m"
+  file="/nethome/users/$USER/outer"
+  mkdir "$file"
+  ensure $file
+
+  echo -e "\033[1;35mCreating folder inner\033[0m"
+  file="/nethome/users/$USER/inner"
+  mkdir "$file" 
+  ensure $file
+
+  folder="/nethome/users/$USER/outer/"
+  file="/nethome/users/$USER/inner"
+  ACL=777
+  file_nesting $folder $file
+  file="$folder${file:22}"
+  ensure $file 
+
+  echo -e "\033[1;35mCreating symlink in dfs\033[0m"
+  file="$file""/sym" 
+  ln -s "/mnt/dfs/$USER/Users/$USER/${folder:22}" "/mnt/dfs/$USER/UserS/$USER/${file:22}"
+  #the above line is here to show that dfs can create symlinks
+  ensure $file
+
+  echo -e "\033[1;35mCreating file this_file.sub\033[0m"
+  file="/nethome/users/$USER/this_file.sub"
+  touch "$file"
+  ensure $file
+
+  permissions $ACL $file
+  file_nesting $folder $file
+  file="$folder${file:22}"
+  ensure $file
+
+  echo -e "\033[1;35mDeleting containing folder\033[0m"
+  rm -r "$folder"
+  file="$folder"
+  ensure $file #expecting $file to not exist
+}
```

Comment:
You should add a test for the bug I just fixed so we can make sure it doesn't come back. To reproduce:

1. Make a file through dfshack
2. Delete it from behind dfshack's back
3. Try to create another file with the same name using dfshack